lib/zuora/api.rb

```diff
@@ -11,24 +11,45 @@ module Zuora
   # @return [Config]
   def self.configure(opts={})
     Api.instance.config = Config.new(opts)
-    HTTPI.logger = opts[:logger]
-    HTTPI.log = opts[:logger] ? true : false
-    Savon.configure do |savon|
-      savon.logger = opts[:logger]
-      savon.log = opts[:logger] ? true : false
+    if Api.instance.config.sandbox
+      Api.instance.sandbox!
+    elsif Api.instance.config.services
+      Api.instance.set_endpoint Api.instance.config.custom_url
     end
   end
 
   class Api
+    I18n.enforce_available_locales = false
+
     # @return [Savon::Client]
-    attr_accessor :client
+    def client
+      @client ||= make_client
+    end
 
     # @return [Zuora::Session]
     attr_accessor :session
 
     # @return [Zuora::Config]
     attr_accessor :config
 
+    # Zuora::API Config options
+    # @return [Hash]
+    attr_accessor :options
+
+    # config object is not available here. Replaced the use of WSDL constant
+    # with a call to fetch_wsdl in the private method
+    # WSDL = if config && config.wsdl_path
+    #          config.wsdl_path
+    #        else
+    #          File.expand_path('../../../wsdl/zuora.a.78.0.wsdl', __FILE__)
+    #        end
```

Comment:
OK, I think it would be best to keep this bit unchanged from the upstream version and reference it in the `#fetch_wsdl`:
```suggestion
    WSDL = File.expand_path('../../../wsdl/zuora.a.78.0.wsdl', __FILE__)
```

Reply:
Resolved here [40ff652](https://github.com/Sage/zuora/pull/18/commits/40ff65273ed3b353a0b77ddd00ab3788778da140)